elastic/kibana
# [Uptime] [Bug] Support KQL "OR" filters

## Summary

Resolves #36855.

As noted in #36855 we have an issue in Uptime where queries like `(monitor.name:test-page OR monitor.name:prod-site)` fail to provide the expected filtering experience. This patch attempts to resolve that issue.

## Testing
0. Start up Heartbeat with several up and several down monitors, run this branch
1. Specify an `OR` filter query
2. Add a status filter
3. Add an additional top-level filter like Location or Scheme
4. Add an additional top-level filter like Port or Name
5. Toggle several of the filters you've added

Ensure that the monitor list you see reflects what you'd expect to see for each set of criteria you've specified.

### Checklist

Use ~~strikethroughs~~ to remove checklist items you don't feel are applicable to this PR.


### For maintainers
